### PR TITLE
C#: Save copy of sln and csproj before applying fixes

### PR DESCRIPTION
**Pull request by neikeq**
This way user can revert in case something goes wrong with the fixes.

### CODE REVIEW

File: modules/mono/editor/GodotTools/GodotTools.Core/FileUtils.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            string backupPathBase = filePath + ".old";
+            string backupPath = backupPathBase;
+
+            const int maxAttempts = 5;
+            int attempt = 0;
+
+            while (File.Exists(backupPath) && attempt < maxAttempts)
+            {
+                backupPath = backupPathBase + "." + (attempt + 1);
```
Comment (aaronfranke) [inline modules/mono/editor/GodotTools/GodotTools.Core/FileUtils.cs:17]: Could be simplified to not have the `+ 1` either by changing the initial value to 1 or by moving the `++` line above this one.

**Review by aaronfranke — COMMENTED** [review]
Should the backup copies be put in `.mono` to avoid them being tracked by version control?

**Comment by neikeq** [comment]
No, it should be next to the original files to be found easily. They can be deleted manually if not needed.

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**